hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/MaintenanceSubCommand.java

```diff
@@ -51,9 +52,20 @@
   @Override
   public void execute(ScmClient scmClient) throws IOException {
     if (hosts.size() > 0) {
-      scmClient.startMaintenanceNodes(hosts, endInHours);
+      List<DatanodeAdminError> errors =
+          scmClient.startMaintenanceNodes(hosts, endInHours);
       System.out.println("Entering maintenance mode on datanode(s):\n" +
           String.join("\n", hosts));
+      if (errors.size() > 0) {
+        for (DatanodeAdminError error : errors) {
+          System.err.println("Error: " + error.getHostname() +": "
+              + error.getError());
+        }
+        // Throwing the exception will cause a non-zero exit status for the
+        // command.
+        throw new IOException(
```

Comment:
I am not sure if this is the best way, but at the moment as I understood GenericCLI we might not have any other way. We may file a follow up JIRA to utilize picocli facilities for returning an exit code instead of throwing an exception here.

Reply:
Yes I agree. I had the same thought and realised that the way we have the GenericCLI setup, it prevents us returning a specific non-zero code. For this soft-failure (some worked some failed), I would like to have returned a specific code.

However, I did not want to get into refactoring Generic CLI as part of this change. The other problem, is that we are somewhat short of unit tests for the CLI commands, so any change we make to that area will need new tests added and some manual validation.

I will raise a Jira with the suggestion tomorrow before committing this. Thanks for the review!